components/notification-sender-config/org.wso2.carbon.identity.notification.sender.tenant.config/src/main/java/org/wso2/carbon/identity/notification/sender/tenant/config/NotificationSenderManagementServiceImpl.java

```diff
@@ -155,62 +153,48 @@ public EmailSenderDTO addEmailSender(EmailSenderDTO emailSender) throws Notifica
 
-        // Set the default publisher name if name is not defined.
-        if (StringUtils.isEmpty(smsSender.getName())) {
-            smsSender.setName(DEFAULT_SMS_PUBLISHER);
-        }
-
-        validateSMSSender(smsSender);
-
-        Optional<Resource> resourceOptional = getPublisherResource(smsSender.getName());
-
-        if (resourceOptional.isPresent()) {
-            throw new NotificationSenderManagementClientException(ERROR_CODE_CONFLICT_PUBLISHER, smsSender.getSender());
+        Map<String, String> properties = smsSender.getProperties();
+        String channel;
+        if (StringUtils.isNotEmpty(properties.get(CHANNEL_TYPE_PROPERTY))) {
+            channel = properties.get(CHANNEL_TYPE_PROPERTY);
+        } else {
+            channel = DEFAULT_HANDLER_NAME;
         }
 
-        Map<String, String> defaultPublisherProperties = getDefaultPublisherProperties(smsSender.getName());
-        // Add the publisher type to the new publisher.
-        defaultPublisherProperties.put(PUBLISHER_TYPE_PROPERTY, SMS_PUBLISHER_TYPE);
-        smsSender.getProperties().putAll(defaultPublisherProperties);
-        Resource smsSenderResource = buildResourceFromSmsSender(smsSender);
-
-        try {
-            /*
-            The input properties will be saved as the attributes of a resource to return to the notification-senders
-            API responses.
-            Also, an event publisher file is generated with the input values and save it as a file of the resource.
-            It is used when loading tenant wise event publisher loading flow.
-             */
-            NotificationSenderTenantConfigDataHolder.getInstance().getConfigurationManager()
-                    .addResource(PUBLISHER_RESOURCE_TYPE, smsSenderResource);
-
-            reDeployEventPublisherConfiguration(smsSenderResource);
-        } catch (ConfigurationManagementException e) {
-            throw handleConfigurationMgtException(e, ERROR_CODE_ERROR_ADDING_NOTIFICATION_SENDER, smsSender.getName());
+        if (NotificationSenderTenantConfigDataHolder.getInstance().getConfigurationHandlerMap().containsKey(channel)) {
+            return NotificationSenderTenantConfigDataHolder.getInstance().getConfigurationHandlerMap()
+                    .get(channel).addSMSSender(smsSender);
+        } else {
+            throw new NotificationSenderManagementClientException(ERROR_CODE_CONFIGURATION_HANDLER_NOT_FOUND);
         }
-        return buildSmsSenderFromResource(smsSenderResource);
     }
 
     @Override
     public void deleteNotificationSender(String senderName) throws NotificationSenderManagementException {
 
-        try {
+        Optional<Resource> resourceOptional = getPublisherResource(senderName);
+        if (!resourceOptional.isPresent()) {
+            throw new NotificationSenderManagementClientException(ERROR_CODE_PUBLISHER_NOT_EXISTS, senderName);
+        }
```

Comment:
```suggestion
        Resource resource = getPublisherResource(senderName).orElseThrow(() -> new NotificationSenderManagementClientException(ERROR_CODE_PUBLISHER_NOT_EXISTS, senderName));
```

Reply:
resolved in https://github.com/wso2-extensions/identity-event-handler-notification/pull/179/commits/9ddb8bd7908408fadef1af4dcc303a3ce60a987c